src/code/z_scene.c

```diff
@@ -362,30 +362,31 @@ void func_80098D80(GlobalContext* globalCtx, SceneCmd* cmd) {
     }
 
     if (cmd->timeSettings.unk_06 != 0xFF) {
-        globalCtx->envCtx.timeIncrement = cmd->timeSettings.unk_06;
+        globalCtx->envCtx.timeSpeed = cmd->timeSettings.unk_06;
     } else {
-        globalCtx->envCtx.timeIncrement = 0;
+        globalCtx->envCtx.timeSpeed = 0;
     }
 
     if (gSaveContext.sunsSongState == SUNSSONG_INACTIVE) {
-        gTimeIncrement = globalCtx->envCtx.timeIncrement;
+        gTimeSpeed = globalCtx->envCtx.timeSpeed;
```

Comment:
I suggest naming the context field in a more specific way, so it doesn't get mistaken as the actual time speed. And based on the usage here, I think it always end up holding the time speed defined in the scene, so maybe we could name it as the "scene time speed" or something along those lines.
```suggestion
        gTimeSpeed = globalCtx->envCtx.sceneTimeSpeed;
```

Similarly I'd suggest naming `R_ENV_TIME_SPEED` around the fact that it doesn't actually get updated or used beyond init, so maybe `R_ENV_TIME_SPEED_INIT`. Or we could just assume it's an old register and name it something like `R_ENV_TIME_SPEED_OLD` to make it less likely to be confused as the real thing